crates/driver/src/domain/competition/solution/settlement.rs

```diff
@@ -386,9 +386,14 @@ impl Gas {
             eth::U256::from_f64_lossy(eth::U256::to_f64_lossy(estimate.into()) * GAS_LIMIT_FACTOR)
                 .into();
 
+        // The block gas limit may fluctuate between blocks (validators trying to
+        // upvote/downvote the limit), thus add a bit margin to not run into
+        // GasLimitExceeded errors.
+        let block_limit = eth::Gas(block_limit.0 * 90 / 100);
```

Comment:
I don't really understand how this solves the issue.
If the previous block had a limit of 30M and the current block has 17M wouldn't we still possibly set the limit to `30M * 0.9 == 27` which is still way bigger than the target of 17M?

At some point I thought this slight change relates to EIP-1559 but that doesn't really apply here, right?
The reasoning was that a node running the old software might increase the base fee for the next block differently than a node running the new software (with the lower gas limit). But that doesn't influence the actual block size, right?

Reply:
The gas limit doesn't go from 17M to 30M in one block but instead validators upvote the gas limit little by little (I can try to find in the yellow paper what's the maximum acceptable increase per block if that helps). But 10% buffer should be plenty.

Reply:
It's 1/1024 per block so 10% seems a bit much. If we use 1% buffer we'd still be good for 10 blocks which seems very sufficient. Will reduce the buffer accordingly.

<img width="336" alt="image" src="https://github.com/cowprotocol/services/assets/1200333/ac97959f-a8b2-4bbb-827b-f12740fa23ac">